src/azure-cli-core/azure/cli/core/adal_authentication.py

```diff
@@ -119,9 +119,13 @@ def set_token(self):
         except requests.exceptions.HTTPError as err:
             logger.debug('throw requests.exceptions.HTTPError when doing MSIAuthentication: \n%s',
                          traceback.format_exc())
-            raise AzureResponseError('Failed to connect to MSI. Please make sure MSI is configured correctly.\n'
-                                     'Get Token request returned http error: {}, reason: {}'
-                                     .format(err.response.status, err.response.reason))
+            err_msg = 'Failed to connect to MSI. Please make sure MSI is configured correctly.\n'
+            if hasattr(err.response, 'status') and hasattr(err.response, 'reason'):
+                err_msg += 'Get Token request returned http error: {}, reason: {}'\
+                    .format(err.response.status, err.response.reason)
```

Comment:
Does this mean `status` and `reason` can be missing? Kindly provide more context.

Reply:
That's right. The `HttpError.response` returned somehow doesn't have `status` and `reason` attribute. In that case, I use the whole response as `err_msg`